framework/src/org/checkerframework/framework/util/AtmLubVisitor.java

```diff
@@ -115,14 +112,8 @@ private AnnotatedTypeMirror lubWithNull(
     private void lubPrimaryAnnotations(
             AnnotatedTypeMirror type1, AnnotatedTypeMirror type2, AnnotatedTypeMirror lub) {
         Set<? extends AnnotationMirror> lubSet;
-        if (visitingUninferedWildcard
-                        && type1.getAnnotations().size() != type1.getAnnotations().size()
```

Comment:
I tried with type1 != type2 as condition but that never got triggered.
#1238 doesn't contain a test case that exercises this branch. I think it is much better to simplify this code and find when this complexity is actually needed.
The variable name 'visitingUninferedWildcard' is also rather poorly chosen - usually we use 'uninferred' for method type arguments that weren't inferred, but here it is used for raw types.